config/samples/dataplane_v1beta1_openstackdataplane_customnetworks.yaml

```diff
@@ -0,0 +1,174 @@
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlane
+metadata:
+  name: openstack-edpm
+spec:
+  deployStrategy:
+    deploy: false
+  nodes:
+    edpm-compute-0:
+      role: edpm-compute
+      hostName: edpm-compute-0
+      ansibleHost: 192.168.1.5
+      node:
+        ansibleVars: |
+          ctlplane_ip: 192.168.1.5
+          internal_api_ip: 172.17.0.101
+          storage_ip: 172.18.0.101
+          tenant_ip: 172.10.0.101
+          external_ip: 172.20.12.76
+          fqdn_internal_api: edpm-compute-1.example.com
+          ansible_ssh_transfer_method: scp
+        ansibleSSHPrivateKeySecret: dataplane-ansible-ssh-private-key-secret
+        ansibleUser: root
+      deployStrategy:
+        deploy: true
+  roles:
+    edpm-compute:
+      env:
+        - name: ANSIBLE_FORCE_COLOR
+          value: "True"
+        - name: ANSIBLE_ENABLE_TASK_DEBUGGER
+          value: "True"
+        - name: ANSIBLE_VERBOSITY
+          value: 4
+        - name: ANSIBLE_HOST_KEY_CHECKING
+          value: false
+      nodeTemplate:
+        managed: false
+        managementNetwork: ctlplane
+        ansibleUser: root
+        ansiblePort: 22
+        ansibleSSHPrivateKeySecret: dataplane-ansible-ssh-private-key-secret
+        networkConfig:
```

Comment:
Wondering if we want to allow users to use built-in templates instead of requiring the whole templates always.

Reply:
It does seem as though there is precedent for setting Ansible vars like this, so I added handling here:
https://github.com/openstack-k8s-operators/dataplane-operator/pull/155/files#diff-506be0d5f31d8a915bfd2ae4c8b305856930dc22e9d8a443813b90e9a8b01356R407

Maybe that would be preferable. Then if there is NOT a `NetworkConfig.Template` file provided, then we will revert to having the user define one of the default templates?